lib/wallet/http.js

```diff
@@ -1482,27 +1482,79 @@ class HTTP extends Server {
       const valid = Validator.fromRequest(req);
       const names = valid.array('names');
       const passphrase = valid.str('passphrase');
-      const broadcast = valid.bool('broadcast', true);
-      const sign = valid.bool('sign', true);
 
       assert(names && names.length > 0, 'Names are required.');
-      assert(broadcast ? sign : true, 'Must sign when broadcasting.');
-
       const options = TransactionOptions.fromValidator(valid);
-      const {mtx, errorMessages} = await req.wallet
-        .createBatchReveal(names, options);
 
-      if (broadcast) {
-        const tx = await req.wallet.sendMTX(mtx, passphrase);
-        return res.json(200, {tx: tx.getJSON(this.network),
-          errors: errorMessages});
+      const revealCache = req.wallet.sendRevealResults;
```

Comment:
Can you split this function up it's doing so many things so it's a bit hard to review

Reply:
fixed ...